pkg/nakadi/nakadi.go

```diff
@@ -62,6 +62,90 @@ func (c *Client) UnconsumedEvents(ctx context.Context, subscriptionID string) (i
 	return unconsumedEvents, nil
 }
 
+type SubscriptionFilter struct {
+	SubscriptionID    string
+	OwningApplication string
+	EventTypes        []string
+	ConsumerGroup     string
+}
+
+func (c *Client) subscriptions(ctx context.Context, filter *SubscriptionFilter, href string) ([]string, error) {
+	endpoint, err := url.Parse(c.nakadiEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if href != "" {
+		endpoint, err = url.Parse(c.nakadiEndpoint + href)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse URL with href: %w", err)
+		}
+	} else {
+		endpoint.Path = "/subscriptions"
+		q := endpoint.Query()
+		if filter.OwningApplication != "" {
+			q.Set("owning_application", filter.OwningApplication)
+		}
+		for _, eventType := range filter.EventTypes {
+			q.Add("event_type", eventType)
+		}
+		if filter.ConsumerGroup != "" {
+			q.Set("consumer_group", filter.ConsumerGroup)
+		}
+		endpoint.RawQuery = q.Encode()
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, endpoint.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to make request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	d, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("[nakadi subscriptions] unexpected response code: %d (%s)", resp.StatusCode, string(d))
```

Comment:
I like the prefix of `[nakadi subsriptions]` prefix in the error string. We should add it to the rest of the error messages in the function as well.

Reply:
added